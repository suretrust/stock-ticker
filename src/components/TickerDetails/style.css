.ticker-details-box {
    margin-top: 85px;
}

.ticker-name {
    font-weight: 500;
    font-size: 23px;
    line-height: 32px;
    color: #00264C;
    margin-top: 34px;
    margin-bottom: 4px;
}

.ticker-sector {
    font-weight: normal;
    font-size: 17px;
    line-height: 30px;
    color: #728BA3;
}

.price-and-color {
    margin-top: 24px;
}

.ticker-price {
    font-weight: 600;
    font-size: 30px;
    line-height: 40px;
    color: #00264C;
}

.ticker-color {
    font-weight: 500;
    font-size: 17px;
    line-height: 24px;
    margin-top: 8px;
}

.ticker-up:after {
    content: url('/src/images/arrow-up.svg');
    margin-left: 15px;
}

.ticker-down:after {
    content: url('/src/images/arrow-down.svg');
    margin-left: 15px;
    display: inline-block;
    transform: rotate(180deg);
}