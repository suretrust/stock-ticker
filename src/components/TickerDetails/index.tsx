import React from 'react';
import { ChangeSinceLastDayProps, TickerDetailsProps } from '../../App';
import { getPercentageChange } from '../../utilities';
import './style.css';

interface Props {
  tickerDetails: TickerDetailsProps | null;
  changeSinceLastDay: ChangeSinceLastDayProps | null;
}

const TickerDetails: React.FC<Props> = ({
  tickerDetails,
  changeSinceLastDay,
}) => {
  if (!tickerDetails) return null;

  const getPriceColor = () => {
    if (!changeSinceLastDay) return;

    const color = isNegative ? 'red' : '#37AD94';

    return { color };
  };

  const isNegative = changeSinceLastDay
    ? getPercentageChange(changeSinceLastDay.o, changeSinceLastDay.c).isNegative
    : null;

  return (
    <div className='ticker-details-box'>
      <img
        className='ticker-logo'
        src={tickerDetails.logo}
        alt={tickerDetails.name}
      />
      <div className='ticker-name'>{tickerDetails.name}</div>
      <div className='ticker-sector'>{tickerDetails.sector}</div>
      {changeSinceLastDay && (
        <div className='price-and-color'>
          <div className='ticker-price'>{changeSinceLastDay.c} USD</div>
          <div
            className={`ticker-color ticker-${isNegative ? 'down' : 'up'}`}
            style={getPriceColor()}
          >
            {
              getPercentageChange(changeSinceLastDay.o, changeSinceLastDay.c)
                .percentage
            }
          </div>
        </div>
      )}
    </div>
  );
};

export default TickerDetails;
