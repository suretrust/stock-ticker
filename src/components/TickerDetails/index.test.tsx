import { render, screen } from '@testing-library/react'
import TickerDetails from '.'

const tickerDetails = {
  logo: 'logo',
  sector: 'test sector',
  url: 'https://url.com',
  name: 'test name',
  symbol: 'exchange',
}

const changeSinceLastDay = {
  o: 125,
  c: 126,
}

describe('TickerDetails component', () => {
  it('renders TickerDetails as fragment', () => {
    const { asFragment } = render(
      <TickerDetails
        tickerDetails={tickerDetails}
        changeSinceLastDay={changeSinceLastDay}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders ticker image with alt text', () => {
    render(
      <TickerDetails
        tickerDetails={tickerDetails}
        changeSinceLastDay={changeSinceLastDay}
      />
    )
    const tickerImage = screen.getByAltText(tickerDetails.name)

    expect(tickerImage).toBeInTheDocument()
  })

  it('renders ticker name', () => {
    render(
      <TickerDetails
        tickerDetails={tickerDetails}
        changeSinceLastDay={changeSinceLastDay}
      />
    )
    const tickerName = screen.getByText(tickerDetails.name)

    expect(tickerName).toBeInTheDocument()
  })

  it('renders ticker sector', () => {
    render(
      <TickerDetails
        tickerDetails={tickerDetails}
        changeSinceLastDay={changeSinceLastDay}
      />
    )
    const tickerSector = screen.getByText(tickerDetails.sector)

    expect(tickerSector).toBeInTheDocument()
  })
})
