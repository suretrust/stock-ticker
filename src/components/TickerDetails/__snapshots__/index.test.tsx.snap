// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerDetails component renders TickerDetails as fragment 1`] = `
<DocumentFragment>
  <div
    class="ticker-details-box"
  >
    <img
      alt="test name"
      class="ticker-logo"
      src="logo"
      width="100"
    />
    <div
      class="ticker-name"
    >
      test name
    </div>
    <div
      class="ticker-sector"
    >
      test sector
    </div>
    <div
      class="price-and-color"
    >
      <div
        class="ticker-price"
      >
        126 USD
      </div>
      <div
        class="ticker-color ticker-up"
        style="color: rgb(55, 173, 148);"
      >
        +1 (0.79%)
      </div>
    </div>
  </div>
</DocumentFragment>
`;
